File: autoit-mediapipe-com/src/binding/tasks/components/processors/classifier_options.cc
#include "binding/tasks/components/processors/classifier_options.h"

using namespace mediapipe::tasks::components::processors;

namespace mediapipe::tasks::autoit::components::processors::classifier_options {
	std::shared_ptr<proto::ClassifierOptions> ClassifierOptions::to_pb2() {
		auto pb2_obj = std::make_shared<proto::ClassifierOptions>();

		if (score_threshold) pb2_obj->set_score_threshold(*score_threshold);
		if (category_allowlist) pb2_obj->mutable_category_allowlist()->Add(category_allowlist->begin(), category_allowlist->end());
		if (category_denylist) pb2_obj->mutable_category_denylist()->Add(category_denylist->begin(), category_denylist->end());
		if (display_names_locale) pb2_obj->set_display_names_locale(*display_names_locale);
		if (max_results) pb2_obj->set_max_results(*max_results);

		return pb2_obj;
	}

	std::shared_ptr<ClassifierOptions> ClassifierOptions::create_from_pb2(const proto::ClassifierOptions& pb2_obj) {
		auto classifier_options = std::make_shared<ClassifierOptions>();
		classifier_options->score_threshold = pb2_obj.score_threshold();
		std::copy(pb2_obj.category_allowlist().begin(), pb2_obj.category_allowlist().end(), classifier_options->category_allowlist->begin());
		std::copy(pb2_obj.category_denylist().begin(), pb2_obj.category_denylist().end(), classifier_options->category_denylist->begin());
		classifier_options->display_names_locale = pb2_obj.display_names_locale();
		return classifier_options;
	}
}

File: src/declarations/tasks/components/processors/classifier_options.js
const ns = "mediapipe::tasks::components::processors";
const nsProgId = ns.replaceAll("::", ".");

module.exports = ({language}) => {
    const progid = `mediapipe.tasks.${ language }.components.processors.classifier_options`;

    return [
        // expose classifier_options properties like in mediapipe python
        [`${ progid }.`, "", ["/Properties"], [
            [`${ ns }::ClassifierOptions`, "ClassifierOptions", "", ["/R", "=this", "/S"]],
        ], "", ""],

        [`struct ${ nsProgId }.ClassifierOptions`, "", [`/progid=${ progid }.ClassifierOptions`, "/Simple", "/DC"], [
            ["std::string", "display_names_locale", "\"en\"", ["/RW"]],
            ["int", "max_results", "-1", ["/RW"]],
            ["float", "score_threshold", "0.0f", ["/RW"]],
            ["std::vector<std::string>", "category_allowlist", "{}", ["/RW"]],
            ["std::vector<std::string>", "category_denylist", "{}", ["/RW"]],
        ], "", ""],
    ];
};
